lib/cfg-lexer.c

```diff
@@ -1127,6 +1127,29 @@ struct _CfgBlock
   CfgArgs *arg_defs;
 };
 
+static void
+_resolve_unknown_blockargs_as_varargs(gpointer key, gpointer value, gpointer user_data)
+{
+  CfgArgs *defs = ((gpointer *) user_data)[0];
+  GString *varargs = ((gpointer *) user_data)[1];
+
+  if (cfg_args_get(defs, key) == NULL)
+    {
+      g_string_append_printf(varargs, "%s(%s) ", (gchar *)key, (gchar *)value);
+    }
+}
+
+static void
+_fill_varargs(CfgBlock *block, CfgArgs *args)
+{
+  GString *varargs = g_string_new("");
+  gpointer user_data[] = { block->arg_defs, varargs };
+
+  cfg_args_foreach(args, _resolve_unknown_blockargs_as_varargs, user_data);
+  cfg_args_set(args, "...", varargs->str);
+  g_string_free(varargs, TRUE);
```

Comment:
I am not 100% sure that using `...` as the syntax to get the variadric argument is such a good idea. It's ok in the declaration and matches the C syntax which is nice.

My problem with using `...` is that it's not very intuitive what it means in the expansion context. Might as well not notice it as I try to understand the configuration.

Some examples where we could turn to:
* Variadric macros in gcc have the arguments in a special define __VA_ARGS__, https://gcc.gnu.org/onlinedocs/cpp/Variadic-Macros.html
* Python nicely encapsulates the variadric part into a variable named by the user
* in php, there's a function called func_get_args()
* in javascript, there's an "arguments" local variable

The original VARARGS idea is not that bad after all, though I would enclose it within double-underscores.

So what do you think about `__VARARGS__` ? It's the same code-wise, but still more catchy to the eye.